p2p/src/cybershake.rs

```diff
@@ -68,19 +73,33 @@ pub struct PublicKey {
 /// An endpoint for sending messages to remote party.
 /// All messages are ordered and encryption key is ratcheted after each sent message.
 pub struct Outgoing<W: io::AsyncWrite + Unpin> {
-    writer: W,
+    writer: Pin<Box<W>>,
     seq: u64,
     kdf: Transcript,
+    buf: Vec<u8>,
+    stage: WriteStage,
+    ciphertext_length: [u8; 2],
+    ciphertext_sent: usize,
+}
+
+enum WriteStage {
+    ReadPlaintext,
```

Comment:
that's a misleading name since we are "writing plaintext". Maybe try simply {Pt, CtLen, Ct}?

Reply:
WriteStage were removed